src/ng/directive/input.js

```diff
@@ -1563,11 +1565,54 @@ function rangeInputType(scope, element, attr, ctrl, $sniffer, $browser) {
     } :
     originalRender;
 
+  if (minAttrType) {
+    ctrl.$validators.min = minAttrType === 'min' && supportsRange ?
+      function noopMinValidator() {
+        // Since all browsers set the input to a valid value, we don't need to check validity
+        return true;
+      } :
+      // ngMin doesn't set the min attr, so the browser doesn't adjust the input value as setting min would
+      function minValidator(modelValue, viewValue) {
+        return ctrl.$isEmpty(viewValue) || isUndefined(minVal) || viewValue >= minVal;
+      };
+
+    setInitialValueAndObserver(minAttrType, 'min', minChange);
+  }
+
+  if (maxAttrType) {
+    ctrl.$validators.max = maxAttrType === 'max' && supportsRange ?
+      function noopMaxValidator() {
+        // Since all browsers set the input to a valid value, we don't need to check validity
+        return true;
+      } :
+      // ngMax doesn't set the max attr, so the browser doesn't adjust the input value as setting max would
+      function maxValidator(modelValue, viewValue) {
+        return ctrl.$isEmpty(viewValue) || isUndefined(maxVal) || viewValue <= maxVal;
+      };
+
+    setInitialValueAndObserver(maxAttrType, 'max', maxChange);
+  }
+
+  function setInitialValueAndObserver(attrType, attrName, changeFn) {
+   var initialAttrValue;
+    if (attrType === attrName) {
+      initialAttrValue = attr[attrName];
+      // Set the actual element attribute so that the browser can adjust the value based on
+      // the max value, which might be interpolated
+      element.attr(attrName, initialAttrValue);
+    }
+
+    // This initalizes the min/max value, so that non-support browsers validate with the correct
+    // values during the initial $render
+    changeFn(initialAttrValue);
+    attr.$observe(attrName, changeFn);
+  }
+
   function minChange(val) {
     if (isDefined(val) && !isNumber(val)) {
       val = parseFloat(val);
     }
-    minVal = isNumber(val) && !isNaN(val) ? val : undefined;
+    minVal = isNumber(val) && !isNaN(val) ? val : 0;
```

Comment:
Good point. I assume that browsers that don't force valid values should not assume min = 0 and max = 100.